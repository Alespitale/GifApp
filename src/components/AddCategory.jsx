import { useState } from "react";
import PropType from "prop-types";

export const AddCategory = ({ addCategory, categoryName }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = ({ target }) => setInputValue(target.value);

  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length < 1)
      return alert("You have to search something");
    if (categoryName.includes(inputValue))
      return alert("You already searched that!");
    addCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search GIF"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propType = {
  addCategory: PropType.func.isRequired,
  categoryName: PropType.string.isRequired,
};
