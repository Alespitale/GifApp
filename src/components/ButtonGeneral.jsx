import styled from "@emotion/styled";

export const ButtonGeneral = styled.button`
  padding: 14px;
  background-color: black;
  font-size: 15px;
  border-radius: 3rem;
  color: white;
  border: 1px solid rgb(97, 32, 158);
  cursor: pointer;
  &:hover {
    color: white;
  }
`;
