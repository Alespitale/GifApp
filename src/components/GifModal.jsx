import Modal from "react-bootstrap/Modal";
import { getDownload } from "../helpers/getDownload";
import PropTypes from "prop-types";

export const GifModal = ({ url, title, show, onHide }) => {
  return (
    <div>
      <Modal className="modal-content" show={show}>
        <Modal.Header className="modal-header">
          <Modal.Title className="modal-title">{title}</Modal.Title>
          <button
            data-testid="close button"
            className="close-btn"
            onClick={onHide}
          >
            x
          </button>
        </Modal.Header>
        <Modal.Body>
          <div>
            <img src={url} alt={title} />
          </div>
          <button
            data-testid="download button"
            className="download-btn"
            onClick={() => getDownload({ url, title })}
          >
            Download
          </button>
        </Modal.Body>
      </Modal>
    </div>
  );
};

GifModal.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  show: PropTypes.bool.isRequired,
  onHide: PropTypes.func.isRequired,
};
