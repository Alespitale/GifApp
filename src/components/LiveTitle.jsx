import "../styles/LiveTitle.css";

export const LiveTitle = () => {
  return (
    <div className="title-container">
      <svg viewBox="0 0 960 100">
        <symbol id="s-text">
          <text textAnchor="middle" x="50%" y="80%">
            Gifinder
          </text>
        </symbol>
        <g className="g-ants">
          <use xlinkHref="#s-text" className="title"></use>
          <use xlinkHref="#s-text" className="title"></use>
          <use xlinkHref="#s-text" className="title"></use>
          <use xlinkHref="#s-text" className="title"></use>
          <use xlinkHref="#s-text" className="title"></use>
        </g>
      </svg>
    </div>
  );
};
