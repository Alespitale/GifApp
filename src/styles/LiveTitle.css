.title-container {
  display: flex;
  justify-content: center;
  margin-top: 30px;
}

.title {
  font: 5.5rem 'GifApp';
  text-align: center;
  text-transform: uppercase;
  stroke-dasharray: 6% 29%;
  stroke-width: 3px;
  stroke-dashoffset: 0%;
  animation: stroke-offset 6s infinite linear;
}

.title:nth-child(1) {
  stroke: #6666ff;
  animation-delay: -1s;
}

.title:nth-child(2) {
  stroke: #0099ff;
  animation-delay: -2s;
}

.title:nth-child(3) {
  stroke: #00ff00;
  animation-delay: -3s;
}

.title:nth-child(4) {
  stroke: #ff3399;
  animation-delay: -4s;
}

.title:nth-child(5) {
  stroke: #f2f205;
  animation-delay: -5s;
}

@keyframes stroke-offset {
  100% {
    stroke-dashoffset: -35%;
  }
}