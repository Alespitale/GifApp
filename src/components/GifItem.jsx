import { useState } from "react";
import { GifModal } from "./GifModal";
import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <div className="card-item">
      <img loading="lazy" src={url} alt={title} onClick={handleShow} />
      <GifModal url={url} title={title} show={show} onHide={handleClose}/>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
