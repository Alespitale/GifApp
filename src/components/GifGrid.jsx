import { useFetchGif } from "../hooks/useFetchGif";
import { GifItem } from "./GifItem";
import { ButtonGeneral } from "./ButtonGeneral";
import PropTypes from 'prop-types'

export const GifGrid = ({ category, clear }) => {
  const { img, loading } = useFetchGif(category);

  return (
    <>
      <div className="category-container">
        <h3>{category}</h3>
        <ButtonGeneral onClick={() => clear(category)}>DELETE</ButtonGeneral>
      </div>
      <div className="card-grid">
        {loading ? (
          <h2>Loading...</h2>
        ) : (
          img.map((imagen) => <GifItem key={imagen.id} {...imagen} />)
        )}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}