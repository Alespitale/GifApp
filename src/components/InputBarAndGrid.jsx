import { AddCategory } from "./AddCategory";
import { ButtonGeneral } from "./ButtonGeneral";
import { GifGrid } from "./GifGrid";
import { useHandleCategories } from "../hooks/useHandleCategories";
import "../styles/InputBarAndGrid.css";

export const InputBarAndGrid = () => {
  const { categories, handleAddCategory, handleResetList, handleDeleteGifs } =
    useHandleCategories();

  return (
    <>
      <div className="input-bar-container">
        <div className="input-bar">
          <AddCategory
            addCategory={handleAddCategory}
            categoryName={categories}
          />
        </div>
        <ButtonGeneral data-testid="reset button" onClick={handleResetList}>
          RESET
        </ButtonGeneral>
      </div>
      {categories.map((category) => (
        <GifGrid key={category} category={category} clear={handleDeleteGifs} />
      ))}
    </>
  );
};
