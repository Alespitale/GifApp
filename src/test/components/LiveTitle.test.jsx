import { render, screen } from "@testing-library/react";
import { LiveTitle } from "../../components/LiveTitle";

describe("Test of LiveTitle", () => {
  test("Should match with snapshot", () => {
    const { container } = render(<LiveTitle />);
    expect(container).toMatchSnapshot();
  });

  test("Should render the text title", () => {
    render(<LiveTitle />);
    expect(screen.getByText("Gifinder").innerHTML).toBe("Gifinder");
  });
});
