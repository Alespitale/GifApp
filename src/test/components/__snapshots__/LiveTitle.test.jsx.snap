// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of LiveTitle Should match with snapshot 1`] = `
<div>
  <div
    class="title-container"
  >
    <svg
      viewBox="0 0 960 100"
    >
      <symbol
        id="s-text"
      >
        <text
          text-anchor="middle"
          x="50%"
          y="80%"
        >
          Gifinder
        </text>
      </symbol>
      <g
        class="g-ants"
      >
        <use
          class="title"
          xlink:href="#s-text"
        />
        <use
          class="title"
          xlink:href="#s-text"
        />
        <use
          class="title"
          xlink:href="#s-text"
        />
        <use
          class="title"
          xlink:href="#s-text"
        />
        <use
          class="title"
          xlink:href="#s-text"
        />
      </g>
    </svg>
  </div>
</div>
`;
